packages/components/src/provider/Provider.js

```
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+import Types from 'prop-types';
+import { createIntl, createIntlCache, RawIntlProvider } from 'react-intl';
+import en from '../../lang/en.json';
+
+const defaultLocale = 'en';
+const cache = createIntlCache();
+
+function createIntlInstance(locale, messages) {
+  return createIntl(
+    {
+      locale,
+      defaultLocale,
+      messages,
+    },
+    cache,
+  );
+}
+
+let currentLocale = defaultLocale;
+
+function Provider({ locale, children }) {
+  const [intl, setIntl] = useState(createIntlInstance(defaultLocale, en));
+
+  useEffect(() => {
+    if (currentLocale !== locale) {
+      currentLocale = locale;
+      (async function changeLanguage() {
+        try {
+          const newMessages = await import(`../../lang/${locale}.json`);
+          setIntl(createIntlInstance(locale, newMessages));
+        } catch (e) {
+          setIntl(createIntlInstance(locale, en));
```

Under what circumstances would this fail? Maybe we should do more than just swallow the error?

Also - maybe we should revert back to `currentLocale` rather than English?

---

it could fail when some unknown values was passed,

good shout, I updated it to handle such cases with logs for errors https://github.com/transferwise/neptune-web/blob/i18y/packages/components/src/provider/Provider.js